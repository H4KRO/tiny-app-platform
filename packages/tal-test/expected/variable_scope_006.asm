main()
  entry:
    FunctionRef       name: root_function_0
    DeclareLocal      name: root_function, mutable: false, hasInitialValue: true
    Pop               inBlock: false
    Literal "a"
    Literal 1
    MakeArray
    Literal 0
    MakeRecord
    Local             name: root_function
    Call
    DeclareLocal      name: a, mutable: false, hasInitialValue: true
    Pop               inBlock: false
    Literal "b"
    Literal 1
    MakeArray
    Literal 0
    MakeRecord
    Local             name: root_function
    Call
    DeclareLocal      name: b, mutable: false, hasInitialValue: true
    Pop               inBlock: false
    Literal 0
    MakeArray
    Literal 0
    MakeRecord
    Local             name: a
    Call
    Pop               inBlock: false
    Literal 0
    MakeArray
    Literal 0
    MakeRecord
    Local             name: b
    Call
root_function_0([object Object])
  entry:
    FunctionRef       name: func_1
    MakeArrayForBlock count: 1
func_1()
  entry:
    Local             name: value
