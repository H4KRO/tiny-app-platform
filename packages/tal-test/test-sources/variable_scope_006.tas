fun root_function(value) {
  () => value
}

let a = root_function("a")
let b = root_function("b")
a()
b()
