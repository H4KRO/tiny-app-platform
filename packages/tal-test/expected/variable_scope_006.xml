<program>
  <declarelocal>
    <function>
      <block>
        <function>
          <local />
        </function>
      </block>
    </function>
  </declarelocal>
  <declarelocal>
    <call>
      <local />
      <literal />
    </call>
  </declarelocal>
  <declarelocal>
    <call>
      <local />
      <literal />
    </call>
  </declarelocal>
  <call>
    <local />
  </call>
  <call>
    <local />
  </call>
</program>
